For_Andrey/task_3.py

```diff
@@ -0,0 +1,255 @@
+                self.mileage -= 1000
+            else:
+                fuel_cost = fuel_consumption * self.mileage * fuel_cost
+                total_fuel_cost += fuel_cost + (fuel_cost * kef)
+                break
+        return total_fuel_cost
+
+    def fuel_cost(self) -> Union[int, float]:
+        """
+        Calculate fuel cost for cars after mileage
+
+        :return:fuel cost with module round
+        """
+        if self.engine == Car.DIESEL_ENGINE_TYPE:
+            return round(self.__fuel_cost_helper(Car.FUEL_CONSUMPTION_DIESEL_ON_HUNDRED_KM, Car.DIESEL_FUEL_COST))
+        else:
+            # For petrol car
+            return round(self.__fuel_cost_helper(Car.FUEL_CONSUMPTION_PETROL_ON_HUNDRED_KM, Car.PETROL_FUEL_COST))
+
+    def number_of_car_refueling(self) -> int:
+        """
+        Calculate refueling depending on gas_tank volume. Use "ceil" because the number of gas stations
+        cannot be rounded down
+
+        :return:number of car refueling
+        """
+        number_of_car_refueling = self.mileage / self.gas_tank
+        time.sleep(0.3)
+        return ceil(number_of_car_refueling)
+
+    def used_car_cost(self) -> int:
+        """
+        Calculate car cost depending on mileage and engine type
+
+        :return:car cost after mileage
+        """
+        if self.engine == Car.DIESEL_ENGINE_TYPE:
+            used_car_cost = Car.NEW_CAR_COST - (Car.DECREASE_VALUE_DIESEL_PER_THOUSAND_KM * (self.mileage // 1000))
+        else:
+            # For petrol car
+            used_car_cost = Car.NEW_CAR_COST - (Car.DECREASE_VALUE_PETROL_PER_THOUSAND_KM * (self.mileage // 1000))
+        if used_car_cost <= 0:
+            return 0
+        return used_car_cost
+
+    def mileage_before_disposal(self) -> Union[int, float]:
+        """
+        Gets used car cost and calculate time used cars before disposal
+
+        :return:mileage before disposal
+        """
+        if self.engine == Car.DIESEL_ENGINE_TYPE:
+            mileage_before_disposal = Car.used_car_cost(self) / (Car.DECREASE_VALUE_DIESEL_PER_THOUSAND_KM / 1000)
+        else:
+            # For petrol car
+            mileage_before_disposal = Car.used_car_cost(self) / (Car.DECREASE_VALUE_PETROL_PER_THOUSAND_KM / 1000)
+        return round(mileage_before_disposal)
+
+    def drive(self):
+        """
+        For start machines at the same time. Use 'semaphore' for limiting the number of cars
+        :return:car arrival message
+        """
+        distance = 0
+        max_car = 10
+        semaphore = threading.BoundedSemaphore(max_car)
+        start = time.time()
+        for i in range(286000):
+            semaphore.acquire()
+            distance += 1000
+            time.sleep(0.3)
+            semaphore.release()
+            if distance >= self.mileage:
+                finish = time.time()
+                print(f'car number {self.car_number} arrived after {int(finish - start)}sec')
+                break
```

Comment:
Use decorator to measure time of execution
https://www.geeksforgeeks.org/timing-functions-with-decorators-python/